patches/server/0977-Add-mob-spawner-ranges-changes-and-API.patch

```diff
@@ -0,0 +1,117 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: DavidTs93 <david.ts93@gmail.com>
+Date: Fri, 7 Apr 2023 19:28:43 +0300
+Subject: [PATCH] Add mob spawner ranges changes and API
+
+
+diff --git a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+index 8d442c5a498ecf288a0cc0c54889c6e2fda849ce..dfd8f1186d01bff7169f687c5771224fbbf138be 100644
+--- a/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
++++ b/src/main/java/io/papermc/paper/configuration/GlobalConfiguration.java
+@@ -276,6 +276,14 @@ public class GlobalConfiguration extends ConfigurationPart {
+                 executor.setMaximumPoolSize(_chatExecutorMaxSize);
+             }
+         }
++
++        public DefaultMobSpawnerRange defaultMobSpawnerRange;
++
++        public class DefaultMobSpawnerRange extends ConfigurationPart {
++            public int horizontal = 4;
++            public int vertical = 1;
++        }
++
+         public int maxJoinsPerTick = 5;
+         public boolean fixEntityPositionDesync = true;
+         public boolean loadPermissionsYmlBeforePlugins = true;
+diff --git a/src/main/java/net/minecraft/world/level/BaseSpawner.java b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+index 31ac0e5ca26c7bdfa9b710d0bb78d846ddf6863e..0e084a9516f6220065c8f2be175f74be1a036277 100644
+--- a/src/main/java/net/minecraft/world/level/BaseSpawner.java
++++ b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+@@ -4,6 +4,8 @@ import com.mojang.logging.LogUtils;
+ import java.util.Optional;
+ import java.util.function.Function;
+ import javax.annotation.Nullable;
++
++import io.papermc.paper.configuration.GlobalConfiguration;
```

Comment:
Don't add this import, instead use the fully qualified class name on the line that needs it. Generally we don't add imports to files we don't create cause it can cause extra merge conflicts when rebasing on upstream changes.